pywbem_mock/_dmtf_cim_schema.py

```diff
@@ -0,0 +1,426 @@
+        self._schema_mof_filename = os.path.join(self.schema_mof_dir,
+                                                 schema_mof_bld_name)
+
+        self.verbose = verbose
+
+        # Setup the local copy of the DMTF CIM schema which may include creating
+        # the schema directory, downloading the schema zip file from the DMTF
+        # web site, and expanding the schema into a subdirectory of the schema
+        # directory.
+        self._setup_dmtf_schema()
+
+    @property
+    def schema_version(self):
+        """
+        :func:`py:tuple`: of 3 integers with major version, minor version, and
+        update version of the DMTF Schema installed. Ex (2, 49, 0) defines DMTF
+        schema version 2.49.0.
+        """
+        return self._schema_version
+
+    @property
+    def schema_version_str(self):
+        """
+        :term:`string`: with the DMTF CIM schema version in the form
+        <major version>.<minor version>.<revison>.
+        """
+        return '%s.%s.%s' % (self.schema_version[0], self.schema_version[1],
+                             self.schema_version[2])
+
+    @property
+    def schema_root_dir(self):
+        """
+        :term:`string`: the directory in which the DMTF CIM Schema
+        MOF is downloaded and expanded.
+        """
+        return self._schema_root_dir
+
+    @property
+    def schema_mof_dir(self):
+        """
+        :term:`string`: Path name of the directory in which the DMTF CIM Schema
+        MOF files are expanded from the downloaded zip file. This includes the
+        expansion of the individual schema MOF files. This property is useful
+        since it can be used as the MOF compiler search path for compiling
+        classes in the DMTF CIM schema.
+        """
+        return self._schema_mof_dir
+
+    @property
+    def schema_mof_filename(self):
+        """
+        :term:`string`: the path for the DMTF CIM schema MOF top level file
+        which includes the pragmas that define all of the files in the DMTF
+        schema. This is the file that the compiler uses if the complete
+        schema is to be compiled. This filename is of the general form::
+
+            cim_schema<schema_version>.mof
+
+        For example::
+
+            cim_schema_2.49.0.mof
+        """
+        return self._schema_mof_filename
+
+    def __str__(self):
+        """
+        Return a short string representation of the
+        :class:`~pywbem_mock.DMTFCIMSchema` object for human consumption.
+        """
+
+        return '%s(schema_version=%s, schema_root_dir=%s, url=%s)' % \
+               (self.__class__.__name__, self.schema_version_str,
+                self.schema_root_dir, self._mof_zip_url)
+
+    def __repr__(self):
+        """
+        Return a string representation of the
+        :class:`~pywbem_mock.DMTFCIMSchema` object that is suitable for
+        debugging.
+        """
+        return '{0}(schema_version={1}, schema_root_dir={2}, ' \
```

Comment:
Any reason we are showing the version tuple in this case instead of the version string? I'm in favor of the version string.

Reply:
I did this because that matches the input parameter.

DISCUSSION

Reply:
ok, fine with me. DONE